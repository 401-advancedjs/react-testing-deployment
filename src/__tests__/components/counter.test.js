import React from 'react';
import Counter from '../../components/counter/counter.js';
import renderer from 'react-test-renderer';

describe( "<Counter/>", () => {
  test('basic rendering', () => {
    const mountedCounter = shallow(<Counter/>);
    expect(mountedCounter.find('a')).toBeTruthy();
  });

  test('testing increment clicks', () => {
    const mountedCounter = mount(<Counter/>);
    const link = mountedCounter.find('a.upClicker');

    link.simulate('click');
    expect(mountedCounter.state('count')).toBe(1);
  });

  test('testing decrement clicks', () => {
    const mountedCounter = mount(<Counter/>);
    const link = mountedCounter.find('a.downClicker');

    link.simulate('click');
    expect(mountedCounter.state('count')).toBe(-1);
  });

  test('rendering follows the snapshot', () => {
    const snapshot = renderer.create(<Counter/>).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
