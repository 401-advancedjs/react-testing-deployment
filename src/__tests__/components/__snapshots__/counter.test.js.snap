// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> rendering follows the snapshot 1`] = `
<section
  className="counter"
>
  <a
    className="downClicker"
    href="/"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="upClicker"
    href="/"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
